eessi/testsuite/tests/apps/osu.py

```diff
@@ -0,0 +1,296 @@
+"""
+This module tests the binary 'osu' in available modules containing substring 'OSU-Micro-Benchmarks'.
+The basic application class is taken from the hpctestlib to which extra features are added.
+"""
+
+import os
+import reframe as rfm
+from reframe.core.meta import parameters
+import reframe.utility.sanity as sn
+
+from hpctestlib.microbenchmarks.mpi.osu import osu_benchmark
+
+from eessi.testsuite import hooks, utils
+from eessi.testsuite.constants import SCALES, TAGS, DEVICE_TYPES
+
+def my_filtering_function():
+    """
+    Filtering function for filtering scales for the pt2pt OSU test
+    """
+    scale_filtered = SCALES.copy()
+    for key in list(SCALES):
+        if(key == '1_core' or key == '4_cores' or
+           SCALES.get(key).get('num_nodes') > 2):
+            scale_filtered.pop(key)
+        elif('node_part' in SCALES.get(key)):
+            if(SCALES.get(key).get('node_part') > 1):
+                scale_filtered.pop(key)
+    return scale_filtered
```

Comment:
removing scales by key is not a good strategy, because if new scales are added that dont fit it will break
    better to select the ones that fit based on resources:
```suggestion
    return [
        k for (k, v) in SCALES.items()
        if v['num_nodes'] * v.get('num_cpus_per_node', 0) == 2
        or (v['num_nodes'] == 2 and v.get('node_part', 0) == 1)
    ]
```

this returns `['2_cores', '1_cpn_2_nodes', '2_nodes']` (with PR 94)

Reply:
This misses 1 node where 1 full node is requested.

Reply:
I have made the change such that `1_node` is now included.